File: 1-two-sum/1-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        
        return HelperTwoSum(nums, target);
    }
    
    public int[] HelperTwoSum(int[] nums, int target) 
    {
        if(nums.Length < 2)
        {
            int[] k = {-1,-1};
            return k;
        }    
        
        
        Dictionary<int, int> dict = new Dictionary<int, int>();
        
        for(int i=0; i<nums.Length; i++)
        {
            int complement = target - nums[i];
            
            if(dict.ContainsKey(complement))
            {
                int[] ans = new int[2];
                ans[0] = dict[complement];
                ans[1] = i;
                
                return ans;
            }
            // IMP condition 
            else if(!dict.ContainsKey(nums[i]))
            {
                dict.Add(nums[i], i);
            }
        }
        
        return null;
    }
}

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        int[] ans = new int[2];
        //int currentSum = 0;
        
        for(int i=0; i<nums.length; i++)
        {
            //currentSum += nums[i];
            
            if(map.containsKey(target - nums[i]))
            {
                ans[1] = i;
                ans[0] = map.get(target - nums[i]);
                return ans;
            }
            
            map.put(nums[i], i);
        }
        
        return new int[2];
    }
}